### PR TITLE
[21832] Prepare for release `3.1.1`

### PR DESCRIPTION
**Pull request by MiguelCompany**
## Description

This bumps the version and adds the release notes for v3.1.1

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- _N/A_: Code snippets related to the added documentation have been provided.
- [x] Documentation tests pass locally.
- _N/A_: Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [ ] CI passes without warnings or errors.

### CODE REVIEW

File: docs/notes/versions.rst [hunk 1]
```diff
@@ -427,9 +427,9 @@ Fast DDS as the core middleware.
...
               - Related version
             * - `Fast DDS Gen <https://github.com/eProsima/Fast-DDS-Gen/>`__
-              - `v4.0.2 <https://github.com/eProsima/Fast-DDS-Gen/releases/tag/v4.0.2>`__
+              - `v4.0.3 <https://github.com/eProsima/Fast-DDS-Gen/releases/tag/v4.0.3>`__
             * - `Fast DDS Gen - IDL parser <https://github.com/eProsima/IDL-Parser/>`__
-              - `v4.0.2 <https://github.com/eProsima/IDL-Parser/releases/tag/v4.0.2>`__
+              - `v4.0.3 <https://github.com/eProsima/IDL-Parser/releases/tag/v4.0.3>`__
             * - `Fast DDS python <https://github.com/eProsima/Fast-DDS-python/>`__
               - `v2.1.0 <https://github.com/eProsima/Fast-DDS-python/releases/tag/v2.1.0>`__
             * - `Shapes Demo <https://github.com/eProsima/ShapesDemo/>`__
```
Comment (Mario-DL) [inline docs/notes/versions.rst:435]: Should we also bump Shapes Demo in this PR to `v3.1.1` ?

Comment (MiguelCompany) [reply docs/notes/versions.rst:435]: Definetely!

Comment (MiguelCompany) [reply docs/notes/versions.rst:435]: 7e9b9f5

**Review by Mario-DL — APPROVED** [review]
LGTM with green CI

**Pull request merged**